contracts/BundleExecutor.sol

```diff
@@ -41,35 +28,76 @@ contract FlashBotsMultiCall {
         _;
     }
 
-    constructor(address _executor) public payable {
+    constructor(ILendingPoolAddressesProvider _addressProvider) FlashLoanReceiverBase(_addressProvider) public payable {
         owner = msg.sender;
-        executor = _executor;
-        if (msg.value > 0) {
-            WETH.deposit{value: msg.value}();
-        }
+        WETH.approve(address(LENDING_POOL), uint(-1));
     }
 
-    receive() external payable {
+    /**
+        This function is called after your contract has received the flash loaned amount
+     */
+    function executeOperation(
+        address[] calldata assets,
+        uint256[] calldata amounts,
+        uint256[] calldata premiums,
+        address initiator,
+        bytes calldata params
+    )
+        external
+        override
+        returns (bool)
+    {
+        uint aaveDebt = amounts[0].add(premiums[0]);
+        uniswapWethFLParams(amounts[0], params, aaveDebt);
+
+        return true;
     }
 
-    function uniswapWeth(uint256 _wethAmountToFirstMarket, uint256 _ethAmountToCoinbase, address[] memory _targets, bytes[] memory _payloads) external onlyExecutor payable {
-        require (_targets.length == _payloads.length);
-        uint256 _wethBalanceBefore = WETH.balanceOf(address(this));
-        WETH.transfer(_targets[0], _wethAmountToFirstMarket);
+    function flashloan(uint256 amountToBorrow, bytes memory _params) external {
+        address receiverAddress = address(this);
+
+        address[] memory assets = new address[](1);
+        assets[0] = WETH_address;
+
+        uint256[] memory amounts = new uint256[](1);
+        amounts[0] = amountToBorrow;
+
+        uint256[] memory modes = new uint256[](1);
+        modes[0] = 0;
+
+        address onBehalfOf = address(this);
+        uint16 referralCode = 161;
+
+        LENDING_POOL.flashLoan(
+            receiverAddress,
+            assets,
+            amounts,
+            modes,
+            onBehalfOf,
+            _params,
+            referralCode
+        );
+    }
+
+    function uniswapWethFLParams(uint256 _amountToFirstMarket, bytes memory _params, uint256 aaveDebt) internal {
```

Comment:
Just like the first one, anyone can perform arbitrary calls, so the "owner" here isn't really that empowered. more a convenience than anything